specification/migrate/resource-manager/Microsoft.Migrate/stable/2019-05-01/migrate.json

```diff
@@ -0,0 +1,3778 @@
+        "nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "GroupResultList": {
+      "type": "object",
+      "description": "List of groups.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "List of groups.",
+          "items": {
+            "$ref": "#/definitions/Group"
+          }
+        }
+      }
+    },
+    "AssessedMachineResultList": {
+      "type": "object",
+      "description": "List of assessed machines.",
+      "properties": {
+        "value": {
+          "description": "List of assessed machines.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AssessedMachine"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "AssessmentResultList": {
+      "type": "object",
+      "description": "List of assessments.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "List of assessments.",
+          "items": {
+            "$ref": "#/definitions/Assessment"
+          }
+        }
+      }
+    },
+    "CloudError": {
+      "x-ms-external": true,
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/CloudErrorBody"
+        }
+      },
+      "description": "An error response from the Azure Migrate service."
+    },
+    "CloudErrorBody": {
+      "x-ms-external": true,
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "An identifier for the error. Codes are invariant and are intended to be consumed programmatically."
+        },
+        "message": {
+          "type": "string",
+          "description": "A message describing the error, intended to be suitable for display in a user interface."
+        },
+        "target": {
+          "type": "string",
+          "description": "The target of the particular error. For example, the name of the property in error."
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudErrorBody"
+          },
+          "description": "A list of additional details about the error."
+        }
+      },
+      "description": "An error response from the Azure Migrate service."
+    },
+    "CollectorBodyAgentSpnProperties": {
+      "type": "object",
+      "properties": {
+        "authority": {
+          "description": "AAD Authority URL which was used to request the token for the service principal.",
+          "type": "string"
+        },
+        "applicationId": {
+          "description": "Application/client Id for the service principal with which the on-premise management/data plane components would communicate with our Azure services.",
+          "type": "string"
+        },
+        "audience": {
+          "description": "Intended audience for the service principal.",
+          "type": "string"
+        },
+        "objectId": {
```

Comment:
Is there a reason that MSIs can't be used for this scenario?  The identity would be represented in Azure by a resource.  https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview

Reply:
Can you expand on that?  MSIs are the way that identities used by a resource should be handled in Azure.  Can you explain how specifying an explicit objectId changes the need to grant that identity permissions to whatever this is acting on?

Reply:
The earlier comment describes why we didn't use MSIs for our S2S between our services. Attaching a mail for that to the mail thread that is going on. Misread your comment earlier. Now for this code, we are storing all these spn details for the agent. These details will be stored in our DB. Now when an agent gets a token from AAD and makes a call to our service we will verify the token against these stored details. We don't know a way where MSIs can work with an agent in our scenario. Moreover, the same pattern is being used in our other service i.e discovery service (which has already been approved in arm review) so we wanted to follow it as a standard paradigm across our services.